### PR TITLE
improve our smoke tests to run on all examples

### PR DESCRIPTION
**Pull request by FredKSchott**
## Changes

- Builds all examples as a part of our smoke test suite
- Runs smoke tests on both windows and ubuntu

## Testing

- Tested in this PR

## Docs

- N/A

### CODE REVIEW

**Comment by natemoo-re** [comment]
The previous code from the `turbo` PR poisoned this cache with some bad code. Will be fixed by merging https://github.com/withastro/astro/pull/2178 and rebasing off of that.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -199,8 +199,7 @@ jobs:
           CI: true
 
       - name: Build
-        run: yarn build
-        working-directory: ${{ matrix.target }}
+        run: node scripts/smoke/index.js
```
Comment (natemoo-re) [inline .github/workflows/ci.yml:202]: Any reason to do a script and not have a specific `test:smoke` command like this?

```bash
turbo run build --scope=docs --scope=www --scope="@example/*"
```

**Review by natemoo-re — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (FredKSchott) [reply .github/workflows/ci.yml:202]: ah good call, I like this

**Comment by FredKSchott** [comment]
Fewer lines of code ftw! @natemoo-re ptal

**Review by natemoo-re — APPROVED** [review]
Code looks good! I'd bump the tests to `node@14.x` to avoid Lit/Svelte issues?

**Review by jonathantneal — APPROVED** [review]

**Pull request merged**